src/main/java/CommandExecutor.java

```diff
@@ -0,0 +1,128 @@
+                new CommandWithArgument(ADD_TODO_COMMAND, ARGUMENT_TASK_DESCRIPTION),
+                new CommandWithFlag(ADD_DEADLINE_COMMAND, ARGUMENT_TASK_DESCRIPTION,
+                        FLAG_DEADLINE_OPTION, FLAG_TASK_TIMESTAMP),
+                new CommandWithFlag(ADD_EVENT_COMMAND, ARGUMENT_TASK_DESCRIPTION,
+                        FLAG_EVENT_OPTION, FLAG_TASK_TIMESTAMP)
+        };
+    }
+
+    public boolean isExit() {
+        return isExit;
+    }
+
+    /**
+     * Validates the given input string before executing it.
+     *
+     * @param inputLine Raw input line to check.
+     */
+    public void execute(String inputLine) {
+        Command command = findCommand(inputLine);
+
+        if (command == null) {
+            System.out.println("[X] Unable to find command.");
+            return;
+        }
+        if (!command.isValidCommandLine(inputLine)) {
+            System.out.printf("[!] Usage: %s\n", command.getUsage());
+            return;
+        }
+
+        try {
+            runCommandUsingInput(command, inputLine);
+        } catch (NumberFormatException err) {
+            System.out.println("[X] Error parsing argument!");
+        }
+    }
+
+    /**
+     * Finds the correct command according to the given input string.
+     *
+     * @param inputLine Raw input line to search.
+     * @return Command that user is trying to run.
+     */
+    private Command findCommand(String inputLine) {
+        for (Command command : commandList) {
+            if (command.isCommand(inputLine)) {
+                return command;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Perform execution of the given command using the given input string.
+     *
+     * @param command   Command that user is trying to run.
+     * @param inputLine Raw input line to read from.
+     */
+    private void runCommandUsingInput(Command command, String inputLine) {
+        String[] commandResults = command.parseCommand(inputLine);
+        Task task;
+
+        switch (command.getCommand()) {
+        case LIST_COMMAND:
+            taskManager.printTaskList();
+            break;
+        case DONE_COMMAND:
+            taskManager.complete(Integer.parseInt(commandResults[ARGUMENT_VALUE_INDEX]));
+            break;
+        case ADD_TODO_COMMAND:
+            task = new Todo(commandResults[ARGUMENT_VALUE_INDEX]);
+            taskManager.addTask(task);
+            break;
+        case ADD_DEADLINE_COMMAND:
+            task = new Deadline(commandResults[ARGUMENT_VALUE_INDEX], commandResults[FLAG_VALUE_INDEX]);
+            taskManager.addTask(task);
+            break;
+        case ADD_EVENT_COMMAND:
+            task = new Event(commandResults[ARGUMENT_VALUE_INDEX], commandResults[FLAG_VALUE_INDEX]);
+            taskManager.addTask(task);
+            break;
+        case END_COMMAND:
+            isExit = true;
+            break;
+        }
+    }
+}

```

Comment:
The use of switch cases and constant names to represent different commands makes the code easy to read and understand! Great Job